libmscore/mscore.cpp

```diff
@@ -79,6 +79,7 @@ bool  MScore::saveTemplateMode = false;
 bool  MScore::noGui = false;
 
 MStyle* MScore::_defaultStyleForParts;
+int     MScore::_unitsValue;
```

Comment:
It is an integer because `pageUnits` is an array and the drop-down lists use an integer index to set and get the value.  It is used more as an integer than as a `QPageLayout::Unit` enum value.

Reply:
`int` tells the reader absolutely nothing about what is the meaning of this variable and which values it can take. If your array's indices are synchronized with that enum then it is better to declare this variable with that enum type. We already do it for many enums, for example, for [`ElementType`](https://github.com/musescore/MuseScore/blob/e410fcf8d7cf16cf26b6ddc43d0bf171914ea614/libmscore/types.h#L34) enum which is synchronized with [the list of elements' names](https://github.com/musescore/MuseScore/blob/e410fcf8d7cf16cf26b6ddc43d0bf171914ea614/libmscore/scoreElement.cpp#L29) to be able to use enum value (converted to `int`) as that array index. If the fact of such a usage is explicitly stated in comments it should be OK to do so.

Reply:
OK, no problem, I'll declare it as `QPageLayout::Unit` and cast as `int` everywhere.

Reply:
The preference will still be an `int` though.  Or should I create a whole new type of preference?

Reply:
I have resolved this by changing only mscore.h.  The private variable is now declared as `QPageLayout::Unit`, but the public `unitsValue()` and `setUnitsValue()` functions take and return `int`.  Those functions do all the type casting.  The rest of the code remains the same.  Easier than I imagined :)